load_balancer.tf

```diff
@@ -64,17 +66,39 @@ locals {
         port           = load_balancer.pool_member_port
         target_address = ipv4_address
         lb             = load_balancer.name
-      }
+        profile        = load_balancer.profile
+      } if(load_balancer.profile != "network-fixed")
     ]
   ])
+
+  nlb_pool_members = flatten([
+    for load_balancer in var.load_balancers :
+    [
+      for server in ibm_is_instance.vsi :
+      {
+        port      = load_balancer.pool_member_port
+        lb        = load_balancer.name
+        target_id = server.id
+        profile   = load_balancer.profile
+      } if(load_balancer.profile == "network-fixed")
+    ]
+  ])
+}
+
+resource "ibm_is_lb_pool_member" "alb_pool_members" {
+  count          = length(local.alb_pool_members)
+  port           = local.alb_pool_members[count.index].port
+  lb             = ibm_is_lb.lb[local.alb_pool_members[count.index].lb].id
+  pool           = element(split("/", ibm_is_lb_pool.pool[local.alb_pool_members[count.index].lb].id), 1)
+  target_address = local.alb_pool_members[count.index].target_address
 }
 
-resource "ibm_is_lb_pool_member" "pool_members" {
-  count          = length(local.pool_members)
-  port           = local.pool_members[count.index].port
-  lb             = ibm_is_lb.lb[local.pool_members[count.index].lb].id
-  pool           = element(split("/", ibm_is_lb_pool.pool[local.pool_members[count.index].lb].id), 1)
-  target_address = local.pool_members[count.index].target_address
+resource "ibm_is_lb_pool_member" "nlb_pool_members" {
```

Comment:
by renaming this, its a breaking change. Please used moved block to prevent breaking change, and please unskip the upgrade test by including the text `UNSKIP UPGRADE TEST` in the next commit

Reply:
done.